framework/src/ble_manager/ble_manager_state.c

```diff
@@ -155,28 +179,26 @@ ble_result_e blemgr_handle_request(blemgr_msg_s *msg)
 			break;
 		}
 
-		sem_init(&scan_queue->countsem, 0, 0);
-		scan_queue->size = SCAN_INFO_BUFFER_SIZE;
-		scan_queue->queue = (trble_scanned_device *)malloc(sizeof(trble_scanned_device) * scan_queue->size);
-		if (scan_queue->queue == NULL) {
-			ret = TRBLE_FAIL;
+		if (ble_queue_init(_event_caller) != BLE_QUEUE_SUCCESS) {
+			ret= TRBLE_FAIL;
```

Comment:
`ret` is initialized with `TRBLE_FAIL`, and not changed before this `break`.
(Changed from line 178, but there is `break`.)
So I think we don't need to set again.
We can reduce the code then.

Reply:
Okay, I will fix it